dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -103,20 +150,74 @@ class CZipExtract :
             if (m_pExtract->m_DirectoryChanged)
                 UpdateDirectory();
 
-            if (!m_pExtract->Extract(m_hWnd, GetDlgItem(IDC_PROGRESS)))
+            if (m_hExtractionThread != INVALID_HANDLE_VALUE)
             {
-                /* Extraction failed, do not go to the next page */
+                CloseHandle(m_hExtractionThread);
+                m_hExtractionThread = INVALID_HANDLE_VALUE;
+            }
+            m_hExtractionThread = CreateThread(
+                NULL, 0,
+                &CExtractSettingsPage::ExtractEntry,
+                this,
+                CREATE_SUSPENDED,
```

Comment:
It's not really useful to do this if `ResumeThread` is the next line anyway.

Reply:
I was advised to do something similar to this when I applied to fix this issue: https://git.reactos.org/?p=reactos.git;a=blob;f=base/setup/reactos/reactos.c;h=55b18a227ea26f71b8bfdb84b17b9e8370d99faf;hb=HEAD#l1200 (see comments on this issue in Jira)

Reply:
That code has a `PropSheet_SetWizButtons` call in between -- so the suspend/resume could possibly make sure the thread doesn't finish and set a different set of buttons before that call.
If ResumeThread follows right after CreateThread, there's no need to suspend.